kclvm/tools/src/LSP/src/state.rs

```diff
@@ -213,81 +213,108 @@ impl LanguageServerState {
-                        let module_cache = self.module_cache.clone();
-                        let scope_cache = self.scope_cache.clone();
-                        move || match url(&snapshot, file.file_id) {
-                            Ok(uri) => {
-                                let mut db = snapshot.db.write();
-                                match compile_with_params(Params {
-                                    file: filename.clone(),
-                                    module_cache,
-                                    scope_cache: None,
-                                    vfs: Some(snapshot.vfs),
-                                }) {
-                                    Ok((prog, diags, gs)) => {
-                                        let current_version = *snapshot
-                                            .opened_files
-                                            .read()
-                                            .get(&file.file_id)
-                                            .unwrap();
-
-                                        // If the text is updated during compilation, the current compilation result will not be output.
-                                        if current_version == version {
-                                            db.insert(
-                                                file.file_id,
-                                                Arc::new(AnalysisDatabase {
-                                                    prog,
-                                                    diags: diags.clone(),
-                                                    gs,
-                                                    version,
-                                                }),
-                                            );
-
-                                            let diagnostics = diags
-                                                .iter()
-                                                .flat_map(|diag| {
-                                                    kcl_diag_to_lsp_diags(diag, filename.as_str())
-                                                })
-                                                .collect::<Vec<Diagnostic>>();
-                                            sender.send(Task::Notify(lsp_server::Notification {
-                                                method: PublishDiagnostics::METHOD.to_owned(),
-                                                params: to_json(PublishDiagnosticsParams {
-                                                    uri,
-                                                    diagnostics,
-                                                    version: None,
-                                                })
-                                                .unwrap(),
-                                            }));
+            match file.change_kind {
+                ChangeKind::Create | ChangeKind::Modify => {
```

Comment:
Could you please put them into another small function and add more unit test cases?